### DISCUSSION TITLE
Drag and drop (files, images, etc)

### DISCUSSION THREAD

**Discussion by guibzo** [cosscom/coss#58] [Suggestions]
Most components in shadcn are in fact composable, which doesn't make much UIs hard or boring to do. 
One component type that I think is really boring and may be complex to create, not only UI but functional wise, is the drop files component. 
Making it UI versatile, with the functionalities being well attended is pretty hard in some layouts.

**Comment by anselumjuju** [comment]
Designers can find building drag-and-drop file upload components tricky and time-consuming. 
I'm already working on these components and have made good progress—it's about halfway done. If @pasqualevitiello and the team aren't already working on them, I'd be happy to contribute my work!

**Comment by pasqualevitiello** [comment]
@juju-anselum Right now, we're not working on any drag-and-drop features. But feel free to send us your work - we're happy to take a look. Just keep in mind we can't promise to use it, since we're trying to keep things consistent across the project. Cheers!

**Comment by anselumjuju** [comment]
I'll try my best to stick within the scope of the project.

**Comment by davidepacilio** [comment]
A set of upload file components has been released today: https://originui.com/file-upload.

**Reply by guibzo** [reply]
Nice! They are really good too!
